tp/tp.py

```diff
@@ -106,13 +108,23 @@ def __init__(self, side_group, weight_factor=1.0, debug=False,
         self.user_data = user_data
         user_data_file = self.conf.get("user_data_file", None)
         if not user_data and user_data_file:
+            self.logger.info("Trying to user data from file...")
             try:
                 with open(user_data_file) as f:
                     self.user_data = f.read()
             except IOError:
                 self.logger.warn("Could not read user data file: %s. Will launch instances without user data.",
                                  user_data_file)
 
+        if not user_data and not user_data_file and self.tapping_group.user_data:
```

Comment:
Why not only checking for `user_data` variable. You should try to set it, if it fails (`not user_data`), try using a different strategy.